src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -99,50 +99,43 @@ private TestSuite Build(Assembly assembly, string assemblyNameOrPath, IDictionar
 
             try
             {
-                if (options.ContainsKey(FrameworkPackageSettings.DefaultTestNamePattern))
-                    TestNameGenerator.DefaultTestNamePattern = options[FrameworkPackageSettings.DefaultTestNamePattern] as string;
+                if (options.TryGetValue(FrameworkPackageSettings.DefaultTestNamePattern, out object defaultTestNamePattern))
+                    TestNameGenerator.DefaultTestNamePattern = defaultTestNamePattern as string;
 
-                if (options.ContainsKey(FrameworkPackageSettings.WorkDirectory))
-                    TestContext.DefaultWorkDirectory = options[FrameworkPackageSettings.WorkDirectory] as string;
+                if (options.TryGetValue(FrameworkPackageSettings.WorkDirectory, out object workDirectory))
+                    TestContext.DefaultWorkDirectory = workDirectory as string;
                 else
                     TestContext.DefaultWorkDirectory = Directory.GetCurrentDirectory();
 
-                if (options.ContainsKey(FrameworkPackageSettings.TestParametersDictionary))
-                {
-                    var testParametersDictionary = options[FrameworkPackageSettings.TestParametersDictionary] as IDictionary<string, string>;
-                    if (testParametersDictionary != null)
-                    {
-                        foreach (var parameter in testParametersDictionary)
-                            TestContext.Parameters.Add(parameter.Key, parameter.Value);
-                    }
-                }
+                if (options.TryGetValue(FrameworkPackageSettings.TestParametersDictionary, out object testParametersDictionary) &&
+                    testParametersDictionary is Dictionary<string, string>)
+                    foreach (var parameter in (IDictionary<string, string>)testParametersDictionary)
+                        TestContext.Parameters.Add(parameter.Key, parameter.Value);
```

Comment:
```suggestion
                if (options.TryGetValue(FrameworkPackageSettings.TestParametersDictionary, out object testParametersDictionary) &&
                    testParametersDictionary is Dictionary<string, string>)
                    {
                        foreach (var parameter in (IDictionary<string, string>)testParametersDictionary)
                            TestContext.Parameters.Add(parameter.Key, parameter.Value);
                    }
```
I know it does not change the code, but I would prefer adding `{` and `}`.